src/decoder/mod.rs

```diff
@@ -494,26 +593,38 @@ impl<R: Read + Seek> DecoderImpl<R> {
     }
 }
 
-impl<R: Read + Seek + Send + Sync + 'static> Decoder<R> {
-    /// Creates a new decoder builder to configure decoder settings.
-    ///
-    /// # Examples
-    /// ```no_run
-    /// use rodio::Decoder;
-    /// use std::fs::File;
-    ///
-    /// let file = File::open("audio.mp3").unwrap();
-    /// let decoder = Decoder::builder()
-    ///     .with_data(file)
-    ///     .with_hint("mp3")
-    ///     .with_gapless(true)
-    ///     .build()
-    ///     .unwrap();
-    /// ```
-    pub fn builder() -> DecoderBuilder<R> {
-        DecoderBuilder::new()
+/// Converts a `File` into a `Decoder` with automatic optimizations.
+///
+/// This implementation:
+/// - Wraps the file in a `BufReader` for better performance
+/// - Sets `byte_len` from file metadata when available, which can improve seeking operations
+///
+/// # Examples
+/// ```no_run
+/// use std::fs::File;
+/// use std::convert::TryFrom;
```

Comment:
is actually in the prelude since rust 2021

Reply:
Removed & done.